auction-server/src/server.rs

```diff
@@ -171,37 +155,9 @@ async fn setup_chain_store(
                     let (chain_id, chain_config, wallet) =
                         (chain_id.clone(), chain_config.clone(), wallet.clone());
                     Some(async move {
```

Comment:
Do we need to use async move?
Maybe we can pass the function directly?

Reply:
filter_map output should be a bunch of `Option<impl Future>` types, so I don't think it's easy to do it, since we want to wrap it inside an Ok((chain_id.clone, ....))